### PR TITLE
only catch & log errors in production

### PR DESCRIPTION
**Pull request by egh**
allows us to examine stacktraces in dev

### CODE REVIEW

File: client/app/initializers/error-handler.coffee [hunk 1]
```diff
@@ -14,15 +14,16 @@ ErrorHandler =
...
-    Ember.onerror = (error) ->
-      if ENV.environment == 'production'
-        if Bugsnag && Bugsnag.notifyException
-          Bugsnag.notifyException(error, "Uncaught Ember Error")
-      else
-        flash.displayMessage 'error', error
-        logError('\n' + error.message + '\n' + error.stack + '\n')
-        throw error
+    # do not handle errors in dev
+    if ENV.environment != 'development'
```
Comment (4thethrillofit) [inline client/app/initializers/error-handler.coffee]: would change this to `isnt`

File: client/app/initializers/error-handler.coffee [hunk 2]
```diff
@@ -14,15 +14,16 @@ ErrorHandler =
...
+      # The global error handler for internal ember errors
+      Ember.onerror = (error) ->
+        if ENV.environment == 'production'
```
Comment (4thethrillofit) [inline client/app/initializers/error-handler.coffee]: use `is` if possible

**Comment by 4thethrillofit** [comment]
so this is saying do nothing if we are in dev mode right? doesn't the `throw error` line do that?

File: client/app/initializers/error-handler.coffee (hunk 2 above)
Comment (egh) [inline client/app/initializers/error-handler.coffee]: I didn't write `ENV.environment == 'production'`, it was there when I got here.

Is this the preferred style? I see `==/!==` and `is/isnt` in the codebase.

**Comment by egh** [comment]
@4thethrillofit rethrowing the exception breaks the stacktrace in ff. not sure about chrome.

**Comment by tonywok** [comment]
Awesome. This has been on my list for a while. :+1:

**Comment by neo-tahi** [comment]
@egh it logs the stack trace from the original error AND it rethrows it, so it should be okay as-is. I think I did that because I wanted us to be able to find and do debugging when something went wrong in development (or debug bugsnag).
— DTrejo

**Comment by egh** [comment]
@neo-tahi That doesn't work though. It breaks the backtrace in firefox and chrome. Here is what it looks like before the change:

screenshot from 2015-03-11 16 22 48

and after:

screenshot from 2015-03-11 16 23 43

In the first, it looks like the error is coming from the error handler.

The same behavior is present in firefox.

**Comment by egh** [comment]
Note that the original error is in line 10 of `assess.js`

**Comment by egh** [comment]
OK, @DTrejo helped me out here. There are a lot of layers of issues going on here. :) Closing this pull request for now.

**Pull request closed without merging**